fplus-lib/src/core/mod.rs

```diff
@@ -3534,6 +3545,81 @@ _The initial issue can be edited in order to solve the request of the verifier.
         Ok(updated_application) // Return the updated ApplicationFile
     }
 
+    pub async fn request_kyc(self, info: KYCRequestedInfo) -> Result<(), LDNError> {
+        let app_model =
+            Self::get_application_model(info.id.clone(), info.owner.clone(), info.repo.clone())
+                .await?;
+
+        let app_str = app_model.application.ok_or_else(|| {
+            LDNError::Load(format!(
+                "Application {} does not have an application field",
+                info.id
+            ))
+        })?;
+        let mut application_file = serde_json::from_str::<ApplicationFile>(&app_str).unwrap();
+        if application_file.lifecycle.state != AppState::Submitted {
+            return Err(LDNError::Load(format!("Application state is {:?}. Expected Submitted", application_file.lifecycle.state)));
+        }
+
+        application_file = application_file.kyc_request();
```

Comment:
```suggestion
        let application_file = application_file.kyc_request();
```

and drop the `mut` in line 3559.

We're not really-really mutating application_file. In `kyc_request` we're cloning things from it, create a new object and overwrite whole application_file with new object. So by dropping `mut` and redeclaring it with new `let` here, we make it more explicit that it's not really being modified, but rewritten completely.